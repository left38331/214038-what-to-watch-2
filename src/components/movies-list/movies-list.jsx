import React from 'react';
import PropTypes from 'prop-types';
import {CardFilm} from "components/card-film/card-film";

export class MoviesList extends React.PureComponent {
  constructor(props) {
    super(props);

    this.films = props.films;
    this.clickTitle = props.clickTitle;
    this.hoverCardHandler = this.hoverCardHandler.bind(this);

    this.state = {
      activeCardFilm: -1
    };
  }

  hoverCardHandler(movieData) {
    this.setState({
      activeCardFilm: movieData.id
    });
  }

  render() {
    return <div className="catalog__movies-list">
      {this.films.map((item) => <CardFilm key={item.id} film={item} clickTitle={this.clickTitle} hoverCardHandler={this.hoverCardHandler}/>)}
    </div>;
  }
}

MoviesList.propTypes = {
  films: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    alt: PropTypes.string.isRequired,
    src: PropTypes.string.isRequired,
    link: PropTypes.string.isRequired,
  })),
  clickTitle: PropTypes.func.isRequired
};
