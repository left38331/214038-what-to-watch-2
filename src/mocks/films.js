export const films = [
  {
    id: 1,
    title: `Fantastic Beats`,
    poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    alt: `Fantastic Beasts: The Crimes of Grindelwald`,
    link: `movie-page.html`,
    preview: `https://www.kinomania.ru/load/n?file=//fs.kinomania.ru/trailers/fbwft-rutlr3_854.mp4`,
    genre: `Fantasy`
  },
  {
    id: 2,
    title: `Bohemian Rhapsody`,
    poster: `img/bohemian-rhapsody.jpg`,
    alt: `Bohemian Rhapsody`,
    link: `movie-page.html`,
    preview: `https://www.kinomania.ru/load/n?file=//fs.kinomania.ru/media/video/c/76/c766c75666111cab4aa6bb75a5424bc0.480.mp4`,
    genre: `Drama`
  },
  {
    id: 3,
    title: `Macbeth.jpg`,
    poster: `img/macbeth.jpg`,
    alt: `Macbeth`,
    link: `movie-page.html`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Drama`
  },
  {
    id: 4,
    title: `Aviator`,
    poster: `img/aviator.jpg`,
    alt: `Aviator`,
    link: `movie-page.html`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genre: `Adventure`
  },
  {
    id: 5,
    title: `War of the worlds`,
    poster: `img/war-of-the-worlds.jpg`,
    alt: `War of the worlds`,
    link: `movie-page.html`,
    preview: `https://www.kinomania.ru/load/n?file=//fs.kinomania.ru/media/video/5/2a/52ab2988c90e81715008ed5189632b8d.480.mp4`,
    genre: `Action`
  },
];
