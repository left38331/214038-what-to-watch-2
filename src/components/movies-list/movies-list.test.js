import React from 'react';
import renderer from 'react-test-renderer';
import {MoviesList} from 'components/movies-list/movies-list';
import {films} from '../../mocks/films';

it(`Render correctly movies-list component`, () => {
  const moviesListComponent = renderer.create(<MoviesList
    films={films}
    hoverCardHandler = {()=>{}}
    clickTitle = {()=>{}}
  />).toJSON();

  expect(moviesListComponent).toMatchSnapshot();
});
