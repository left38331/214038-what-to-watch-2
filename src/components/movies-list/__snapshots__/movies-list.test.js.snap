// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly movies-list component 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <a
      href="/films/1"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <video
          height="175"
          poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
          src="https://www.kinomania.ru/load/n?file=//fs.kinomania.ru/trailers/fbwft-rutlr3_854.mp4"
          width="280"
        />
      </div>
    </a>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/films/1"
        onClick={[Function]}
      >
        Fantastic Beats
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <a
      href="/films/2"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <video
          height="175"
          poster="img/bohemian-rhapsody.jpg"
          src="https://www.kinomania.ru/load/n?file=//fs.kinomania.ru/media/video/c/76/c766c75666111cab4aa6bb75a5424bc0.480.mp4"
          width="280"
        />
      </div>
    </a>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/films/2"
        onClick={[Function]}
      >
        Bohemian Rhapsody
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <a
      href="/films/3"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <video
          height="175"
          poster="img/macbeth.jpg"
          src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
          width="280"
        />
      </div>
    </a>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/films/3"
        onClick={[Function]}
      >
        Macbeth.jpg
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <a
      href="/films/4"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <video
          height="175"
          poster="img/aviator.jpg"
          src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
          width="280"
        />
      </div>
    </a>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/films/4"
        onClick={[Function]}
      >
        Aviator
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <a
      href="/films/5"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <video
          height="175"
          poster="img/war-of-the-worlds.jpg"
          src="https://www.kinomania.ru/load/n?file=//fs.kinomania.ru/media/video/5/2a/52ab2988c90e81715008ed5189632b8d.480.mp4"
          width="280"
        />
      </div>
    </a>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/films/5"
        onClick={[Function]}
      >
        War of the worlds
      </a>
    </h3>
  </article>
</div>
`;
